papis/docmatcher.py

```diff
@@ -169,7 +191,13 @@ def parse(cls, search: Optional[str] = None) -> List[ParseResult]:
         """
         if search is None:
             search = cls.search
-        cls.parsed_search = parse_query(search)
+        parsed_search = parse_query(search)
+
+        if check_syntax(parsed_search):
+            cls.parsed_search = parsed_search
+        else:
+            cls.parsed_search = []
```

Comment:
I think this sort of check should be just part of the tests. Normally `pyparsing` would throw an error if it does not manage to parse something and our own post-processing should be relatively simple.

Reply:
I guess this is a somewhat misleading function name. `check_syntax` actually checks for logical consistency of a query that is not controlled by `pyparsing`. So it catches everything that cannot be evaluated as an expression of boolean values combined with operators such as `True or and False` or `True or or False` or `True and False or` etc. Btw. I put this in `parse` because otherwise `return_if_match` throws an error for each document in the library.